### PR TITLE
Force update validation

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/3001
Fixes https://github.com/openshift/origin/issues/2822

Adds missing validateUpdate methods.  Also adds unit test for ensuring that validationUpdate methods exists for API objects that need them.

@liggitt or @derekwaynecarr ptal

@smarterclayton Approval and either accept or rejected the unit test commit.  Relatively minor changes in `master.go` for significant gain.

### CODE REVIEW

File: pkg/api/validation/coverage_test.go [hunk 1]
```diff
@@ -62,6 +58,7 @@ func TestCoverage(t *testing.T) {
 
 			if !allowed {
 				t.Errorf("%v is not registered.  Look in pkg/api/validation/register.go.", apiType)
+				return
```
Comment (liggitt) [inline pkg/api/validation/coverage_test.go]: not continue?

Comment (deads2k) [inline pkg/api/validation/coverage_test.go]: > not continue?

Debugging line.  Removed.

File: pkg/build/registry/buildconfig/strategy_test.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ func TestBuildConfigStrategy(t *testing.T) {
 		t.Errorf("BuildConfig should not allow create on update")
 	}
 	buildConfig := &buildapi.BuildConfig{
-		ObjectMeta: kapi.ObjectMeta{Name: "config-id", Namespace: "namespace"},
+		ObjectMeta: kapi.ObjectMeta{Name: "config-id", Namespace: "namespace", ResourceVersion: "foo"},
```
Comment (liggitt) [inline pkg/build/registry/buildconfig/strategy_test.go]: don't set ResourceVersion until after testing the Create call

Comment (deads2k) [inline pkg/build/registry/buildconfig/strategy_test.go]: There is no create call.  After `.PrepareForCreate`?

Comment (liggitt) [inline pkg/build/registry/buildconfig/strategy_test.go]: yeah, sorry... just prior to ValidateUpdate

File: pkg/route/registry/route/rest_test.go [hunk 3]
```diff
@@ -253,7 +265,7 @@ func TestUpdateRouteOK(t *testing.T) {
 	}
 
 	obj, created, err := storage.Update(kapi.NewDefaultContext(), &api.Route{
-		ObjectMeta:  kapi.ObjectMeta{Name: "bar"},
+		ObjectMeta:  kapi.ObjectMeta{Name: "bar", Namespace: kapi.NamespaceDefault, ResourceVersion: "foo"},
```
Comment (liggitt) [inline pkg/route/registry/route/rest_test.go]: shouldn't need the namespace... BeforeUpdate should populate it

Comment (liggitt) [inline pkg/route/registry/route/rest_test.go]: oh... routes don't use genericetcd

Comment (liggitt) [inline pkg/route/registry/route/rest_test.go]: oh, but they do call ValidNamespace, so it still shouldn't be needed

Comment (deads2k) [inline pkg/route/registry/route/rest_test.go]: > oh... routes don't use genericetcd

Different issue/pull for that change.

**Comment by liggitt** [comment]
remove namespace set in route test, then LGTM

**Comment by deads2k** [comment]
comments addressed.

@smarterclayton  approval?  You really want the first commit at least.

**Comment by smarterclayton** [comment]
Approved.  Master wasn't as ugly as I expected.

**Comment by deads2k** [comment]
> Approved. Master wasn't as ugly as I expected.

Me either, though I only got half of what I wanted.  [merge]

**Pull request merged**